Jenkinsfile

```diff
@@ -51,26 +52,34 @@ pipeline {
     }
 
     stage('Release') {
-      when {
-        tag pattern: "[0-9]+\\.[0-9]+\\.[0-9]+", comparator: "REGEXP"
-      }
-
       agent {
         docker {
           image "us.gcr.io/logdna-k8s/python:3.7-ci"
           customWorkspace "${PROJECT_NAME}-${BUILD_NUMBER}"
         }
       }
+
       steps {
-        witCredentials([
-          usernamePassword(
-            credentialsId: 'pypi-username-password',
-            usernameVariable: 'TWINE_USERNAME',
-            passwordVariable: 'TWINE_PASSWORD'
+        withCredentials([
+          string(
+            credentialsId: 'pypi-token',
+            variable: 'PYPI_TOKEN',
+          ),
+          string(
+            credentialsId: 'github-api-token',
+            variable: 'GITHUB_TOKEN',
```

Comment:
Is there a better way to do this? `string` feels dangerous

Reply:
The  [main way](https://www.jenkins.io/doc/book/pipeline/jenkinsfile/#secret-text) to handle both of these to replace this `withCredentials([]){}` block with:

```
environment {
  GITHUB_TOKEN = credentials('github-api-token')
  PYPI_TOKEN = credentials('pypi-token')
}
```

Reply:
I'm not entirely sure that works with a docker context

https://issues.jenkins.io/browse/JENKINS-42369

Reply:
Nevermind, It looks like everything is picked up, I think this jira is in reference to a slightly different problem